qiskit/aqua/algorithms/phase_estimators/hamiltonian_pe.py

```diff
@@ -0,0 +1,174 @@
+        qc.global_phase = global_phase
+        return qc
+
+
+class HamiltonianPE(PhaseEstimator):
+    r"""Run the Quantum Phase Estimation algorithm to find the eigenvalues of a Hermitian operator.
+
+    This class is nearly the same as :class:`~qiskit.aqua.algorithms.PhaseEstimator`, differing only
+    in that the input in that class is a unitary operator, whereas here the input is a Hermitian
+    operator from which a unitary will be obtained by scaling and exponentiating. The scaling is
+    performed in order to prevent the phases from wrapping around :math:`2\pi`. This class uses and
+    works together with :class:`~qiskit.aqua.algorithms.PhaseEstimationScale` to manage scaling the
+    Hamiltonian and the phases that are obtained by the QPE algorithm. This includes setting, or
+    computing, a bound on the eigenvalues of the operator, using this bound to obtain a scale
+    factor, scaling the operator, and shifting and scaling the measured phases to recover the
+    eigenvalues.
+
+    Note that, although we speak of "evolving" the state according the the Hamiltonian, in the
+    present algorithm, we are not actually considering time evolution. Rather, the role of time is
+    played by the scaling factor, which is chosen to best extract the eigenvalues of the
+    Hamiltonian.
+
+    A few of the ideas in the algorithm may be found in Ref. [1].
+
+    **Reference:**
+
+    [1]: Quantum phase estimation of multiple eigenvalues for small-scale (noisy) experiments
+         T.E. O'Brien, B. Tarasinski, B.M. Terhal
+         https://arxiv.org/abs/1809.09697
+    """
+    def __init__(self,
+                 num_evaluation_qubits: int,
+                 hamiltonian: OperatorBase,
+                 evolution: Optional[Union[EvolutionBase, TempPauliEvolve]] = None,
+                 state_preparation: Optional[QuantumCircuit] = None,
+                 bound: Optional[float] = None,
+                 quantum_instance: Optional[Union[QuantumInstance, BaseBackend]] = None) -> None:
+        """
+        Args:
+            num_evaluation_qubits: The number of qubits used in estimating the phase. The
+                                   phase will be estimated as a binary string with this many
+                                   bits.
+            hamiltonian: a Hamiltonian or Hermitian operator
+            evolution: An evolution object that generates a unitary from `hamiltonian`.
+            state_preparation: The circuit that prepares the state whose eigenphase will be
+                                 measured.  If this parameter is omitted, no preparation circuit
+                                 will be run and input state will be the all-zero state in the
+                                 computational basis.
+            bound: An upper bound on the absolute value of the eigenvalues of
+                `hamiltonian`. If omitted, and `hamiltonian` is a Pauli sum, then a bound will be
```

Comment:
If omitted and hamiltonian is not a pauli sum what happens?  Ok I see it tries the from_pauli_sum which raises a Value error. Maybe the fact that a ValueError is raised in this case should be noted in the docstring here in a Raises block